backend/Origam.Server/IdentityServerGui/Account/ExternalController.cs

```diff
@@ -194,37 +200,51 @@ public async Task<IActionResult> Callback()
             return Redirect(returnUrl);
         }
         
-        private (IOrigamUser user, string provider, string providerUserId, IEnumerable<Claim> claims) FindUserFromExternalProvider(AuthenticateResult result)
+        private 
+            (IOrigamUser user, 
+            string provider, 
+            string providerUserId, 
+            IEnumerable<Claim> claims) FindUserFromExternalProvider(
+                AuthenticateResult result)
         {
             var externalUser = result.Principal;
-
-            // try to determine the unique id of the external user (issued by the provider)
-            // the most common claim type for that are the sub claim and the NameIdentifier
-            // depending on the external provider, some other claim type might be used
-            var userIdClaim = externalUser.FindFirst(JwtClaimTypes.Subject) ??
-                              externalUser.FindFirst(ClaimTypes.NameIdentifier) ??
-                              throw new Exception("Unknown userid");
-
-            // remove the user id claim so we don't include it as an extra claim if/when we provision the user
+            // try to determine the unique id of the external user
+            // (issued by the provider)
+            // the most common claim type for that are the sub claim and
+            // the NameIdentifier depending on the external provider,
+            // some other claim types might be used
+            var userIdClaim 
+                = externalUser.FindFirst(JwtClaimTypes.Subject) 
+                  ?? externalUser.FindFirst(ClaimTypes.NameIdentifier) 
+                  ?? throw new Exception("Unknown userid");
```

Comment:
The text of the exception should be more descriptive (what happened and what should be done to mitigate).